### ISSUE TITLE
RFC: Tackling the project distribution problem

### ISSUE THREAD

**Issue by ProLoser** [bower/bower#968]
So I've been privy to quite a few discussions on how distribution, packaging, etc is handled by bower, which has the rare bonus of (usually) needing the assets to be 'built' before distribution.

I've tried to scrape together every idea and solution I could find for making this behavior seamless:
- Committing dist files
- Creating a dist branch
- Creating secondary dist repos
- Hooking into the Github API for releases
- Playing with git export ignore configurations
- Setting up travis to build and commit builds back into repo/branch/dist-repo, etc

All of these present huge annoyances and frustrations:
- Keeping 2 repos for each single project
- Committing built files
- Committed built files being changed when hacking (etc)
- The extra tedium of setting up locally and firing off release scripts
- The extra tedium of keeping multi branches/repos/whatever build-to-src in sync
- Being able to switch from dist files to src files when you want to hack on one of your subpackeges

One solution I liked the most was using Travis to run tests, generate a build, and commit it back for us. This meant that devs didn't have to have release-tools or permissions properly configured to cut a release. But it still cluttered project history and filetree, etc.

So I was thinking, what if a package distribution solution had the ability to run builds? I mean, we do it in TravisCI, so better yet:
# TL;DR:

What if we combined a testing solution (like TravisCI) and a package distribution solution (like Bower) so that the same server could run tests, if they pass, generate a build, and then distribute the generated build?

**All you have to do to cut a release is tag the source code**

_I realize this will in all likeliness get shot down, but I thought I'd throw it out there._
